modmesh/apputil.py

```diff
@@ -57,46 +58,68 @@ class AppEnvironment:
     :ivar locals:
         The local namespace of the application.
     """
-    def __init__(self, name):
+
+    # default config
+    _config = {
+        'appEnvName': None,
+        'redirectStdOutFile': 'stdout.txt',
+        'redirectStdErrFile': 'stderr.txt',
+    }
+
+    def __init__(self, config=None):
         self.globals = {
             # Give the application an alias of the top package.
             'mm': importlib.import_module('modmesh'),
             'appenv': self,
         }
         self.locals = {}
-        self.name = name
         # Each run of the application appends a new environment.
-        environ[name] = self
+        environ[config['appEnvName']] = self
 
-    def run_code(self, code):
-        exec(code, self.globals, self.locals)
+        # replace the config value; otherwise keep the default value
+        for k, v in config.items():
+            if v is not None:
+                self._config[k] = v
 
-
-def get_appenv(name=None):
-    if None is name:
+    def run_code(self, code):
+        with open(self._config['redirectStdOutFile'], 'w') as f1:
```

Comment:
Please use a single level of nesting.  In this revision, you descend by 4 levels before the try/except block, and it is too deep for readability and maintainability.

Reply:
It does not look right to open a file in Python for consumption in C++.  By doing this, when accessing the file from C++, it could wind back in Python for I/O.  It is hard to maintain and slow.

Please open a file in C++ and then wrap it to Python instead.